DevEdu.API/DevEdu.API/Controllers/LessonController.cs

```diff
@@ -1,57 +1,67 @@
-﻿using DevEdu.API.Models.InputModels;
+using DevEdu.API.Models.InputModels;
 using Microsoft.AspNetCore.Mvc;
+using DevEdu.DAL.Repositories;
+using AutoMapper;
+using DevEdu.DAL.Models;
 
 namespace DevEdu.API.Controllers
 {
     [ApiController]
     [Route("api/[controller]")]
     public class LessonController : Controller
     {
-        public LessonController()
-        {
+        private readonly IMapper _mapper;
+        private readonly ILessonRepository _lessonRepository;
 
+        public LessonController(IMapper mapper, ILessonRepository lessonRepository)
+        {
+            _mapper = mapper;
+            _lessonRepository = lessonRepository;
         }
 
         // api/lesson
         [HttpPost]
         public string AddLesson([FromBody] LessonInputModel inputModel)
         {
-            return $"Date {inputModel.Date} TeacherComment {inputModel.TeacherComment}  TeacherId {inputModel.TeacherId}";
+            var dto = _mapper.Map<LessonDto>(inputModel);
+            return _lessonRepository.AddLesson(dto).ToString();
         }
 
         // api/lesson/{id}
         [HttpDelete("{id}")]
         public string DeleteLesson(int id)
         {
+            _lessonRepository.DeleteLesson(id);
             return $"id {id}";
         }
 
         // api/lesson/{lessonId}/comment/{commentId}
         [HttpPost("{lessonId}/comment/{commentId}")]
         public string AddLessonComment(int lessonId, int commentId)
         {
-            return $"lessonId {lessonId} commentId {commentId}";
+            return _lessonRepository.AddCommentToLesson(lessonId, commentId).ToString();
```

Comment:
what are we converting to a string here?)
I think this method just to be a void